### PR TITLE
Add S3 reusable code

### PR DESCRIPTION
**Pull request by noelia-mari**
This PR introduces the "storage" module.

This module is for AWS S3 communication.

### CODE REVIEW

File: xocto/storage/files.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import os
+from typing import IO
+
+XLRD_FLOAT_TYPE = 2
+
+
+def size(file: IO) -> int:
```
Comment (jbkkd) [inline xocto/storage/files.py:7]: This isn't the entire `files.py` file. 

We should move the entire file, not just parts of it

Comment (jbkkd) [reply xocto/storage/files.py:7]: Also, we should move the `test_files.py` file

File: tests/test_storage.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import os
```
Comment (jbkkd) [inline tests/test_storage.py:1]: We should move all the tests from `test_storage.py`

**Review by jbkkd — CHANGES_REQUESTED** [review]

File: tests/test_storage.py (hunk 2 above)
Comment (noelia-mari) [reply tests/test_storage.py:1]: I didn't know that file existed, I just created this one. I'll move all existing tests 🙂

File: xocto/storage/storage.py [hunk 3]
```diff
@@ -0,0 +1,1574 @@
...
+)
+
+
+def _should_raise_error_on_existing_files() -> bool:
+    """
+    Check if we should error when a file is being overwritten without explicit overwrite=True.
+    """
+    # Defaults to `False` for backwards compatibility. TODO(martin): flip to `True` when we're
+    # confident it won't break everything.
+    return False
```
Comment (cgl) [inline xocto/storage/storage.py:74]: # TO DO
Fix the functionality here.

**Comment by cgl** [comment]
LGTM!

**Review by jbkkd — APPROVED** [review]

**Review by cgl — APPROVED** [review]

File: setup.py [hunk 4]
```diff
@@ -38,13 +38,27 @@
         "pact-python>=1.6.0",
     ],
     extras_require={
-        "dev": ["wheel==0.37.1", "twine==4.0.0", "black==22.3.0", "isort==5.10.1"],
+        "dev": [
+            "wheel==0.37.1",
+            "twine==4.0.0",
+            "black==22.3.0",
+            "isort==5.10.1",
+            "boto3==1.24.94",
```
Comment (codeinthehole) [inline setup.py:46]: Isn't boto now a production requirement, not a dev one?

File: xocto/storage/storage.py [hunk 5]
```diff
@@ -0,0 +1,1563 @@
...
+    store_class = import_string(settings.EMAIL_STORAGE_BACKEND)
+    return store_class(bucket_name)
+
+
+def fileserver_store(use_date_in_key_path: bool = True) -> BaseS3FileStore:
+    return store(settings.S3_FILESERVER_BUCKET, use_date_in_key_path)
+
+
+def flows_outbound_store() -> BaseS3FileStore:
+    return store(settings.S3_PRODUCTION_FLOWS_OUTBOUND)
```
Comment (codeinthehole) [inline xocto/storage/storage.py:1454]: These factory functions are quite specific to KrakenCore - am not sure this should be in an open-source repo.

**Review by codeinthehole — COMMENTED** [review]
Couple of retro-comments.

**Pull request merged**